base/REPLCompletions.jl

```diff
@@ -130,8 +130,12 @@ function complete_path(path::AbstractString, pos)
             push!(matches, id ? file * (@windows? "\\\\" : "/") : file)
         end
     end
-    matches = UTF8String[replace(s, r"\s", "\\ ") for s in matches]
-    return matches, nextind(path, pos - sizeof(prefix) - length(matchall(r" ", prefix))):pos, length(matches) > 0
+    matches = UTF8String[replace(s, r" ", "\\ ") for s in matches]
```

Comment:
> It do not seem to do any difference since file names cannot contain tabs, return carriage and new line.

posix filenames can contain any character except `/` (i haven't looked at the rest of this PR to see how that fits in)